src/Type/Generic/GenericObjectType.php

```diff
@@ -121,6 +122,11 @@ private function isSuperTypeOfInternal(Type $type, bool $acceptsContext): Trinar
 			return $nakedSuperTypeOf;
 		}
 
+		if ($type instanceof ThisType && $type->getClassName() === $this->getClassName()) {
+			$type = $type->getStaticObjectType();
+			$nakedSuperTypeOf = TrinaryLogic::createYes();
```

Comment:
Initially I tried making `parent::isSuperTypeOf($type);` return `TrinaryLogic::createYes()`, but thats no possible because that would make the `TypeCombintor` combine types like `$this(TraitInstanceOf\FinalTrait2Class)&TraitInstanceOf\FinalTrait2Class` into `$this(TraitInstanceOf\FinalTrait2Class)` which makes other tests fail.

therefore I figured special casing here seems to be the way to go.